src/components/Editor/index.tsx

```diff
@@ -264,7 +272,7 @@ const Editor = () => {
 
     // Synchronize `isPrettifiedRef` with `isPrettified`.
```

Comment:
similar to https://github.com/y-scope/yscope-log-viewer/pull/286/files#r2108340456 , we may not no long need to copy states into refs. instead, we can directly read them from the Zustand stores.